### ISSUE TITLE
Add SSL and Host support for streams

### ISSUE THREAD

**Issue by simqui** [NginxProxyManager/nginx-proxy-manager#1829]
**Is your feature request related to a problem? Please describe.**

If i want to ex: 
Add a host stream for TCP with SSL support to proxy DoT( Dns-over-TLS) this is not possible right now.

This requires modules: ngx_stream_map_module and ngx_stream_ssl_preread_module
to be implemented.

Also a feature in the WebUI to be able to add streams with SSL and host support.

**Describe the solution you'd like**

Add the ability to use hosts redirect and SSL with a TCP stream proxy.

Include these modules in proxy manager: 
https://nginx.org/en/docs/stream/ngx_stream_ssl_preread_module.html
https://nginx.org/en/docs/stream/ngx_stream_map_module.html

**Describe alternatives you've considered**

Only alternative is to forward TCP 853 directly, and not use NGINX Proxy Manager.

**Additional context**

These modules are needed for this to work and does need to be implemented.

https://nginx.org/en/docs/stream/ngx_stream_ssl_preread_module.html
https://nginx.org/en/docs/stream/ngx_stream_map_module.html

**Comment by HellStorm666** [comment]
Any timeline on this idea to be implemented?

**Comment by razza-guhl** [comment]
+1
Want to stream DoT with SSL termination on Nginx Proxy Manager

**Comment by kkoenen** [comment]
+1 

Forwarding a specific port towards a backend server now is  only possible without SSL (using stream).

**Comment by cybershoe** [comment]
+1. Want to terminate TLS for MQTT at the perimeter with mTLS to keep unauthenticated traffic away from the origin servers.

**Comment by lesca** [comment]
+1 SSL is everywhere. Please add support.

**Comment by mariusberget92** [comment]
+1. :)

**Comment by unnikked** [comment]
+1

**Comment by m0d3rnX** [comment]
+1

**Comment by scharez** [comment]
Any updates? :)

**Comment by countersoda** [comment]
+1

**Comment by kantmn** [comment]
+1 
would love to use this for setup with DoT support

**Comment by scharez** [comment]
Is that feature already planned?

**Comment by sorcerb** [comment]
Just installed this to try, and I can't use stream with SSL for rabbitMq.
I see that people are waiting and after 2 years, I will look for a another product...

**Comment by gadget78** [comment]
can this not be added, by simply adding a custom stream .conf .. 

like pointed out in this comment ?
https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3628

**Comment by sorcerb** [comment]
Yes, but for example [0xJacky](https://github.com/0xJacky) have custom conf from UI out of box
. I use it for my server.

**Comment by XtremeOwnageDotCom** [comment]
Bump.

**Comment by braian87b** [comment]
https://github.com/NginxProxyManager/nginx-proxy-manager/pull/3789

**Comment by kkoenen** [comment]
this should not be stale, imho.

**Comment by scharez** [comment]
> this should not be stale, imho.

It´s already included: https://github.com/NginxProxyManager/nginx-proxy-manager/releases/tag/v2.12.3
